src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -101,6 +112,23 @@ public Task<IHtmlContent> DisplayAsync(dynamic shape)
             throw new ArgumentException("DisplayAsync requires an instance of IShape");
         }
 
+        /// <summary>
+        /// Renders a shape when a specified feature is enabled.
+        /// </summary>
+        /// <param name="shape">The shape.</param>
+        /// <param name="featureId">The feature Id in which <paramref name="shape"/> will be rendered when its enabled.</param>
+        public async Task<IHtmlContent> DisplayAsync(dynamic shape, string featureId)
```

Comment:
Remove ?

Reply:
This is WIP, related to https://github.com/OrchardCMS/OrchardCore/discussions/12014 I will change it once we agreed on a proper solution for render a shape conditionally